apps/site/src/app/apply/sections/Components/SimpleRadio.tsx

```diff
@@ -0,0 +1,59 @@
+import RequiredAsterisk from "./RequiredAsterisk";
+
+interface TextfieldProps {
+	name: string;
+	values: Array<{
+		id: string;
+		labelText: string;
+		inputValue: string;
+	}>;
+	title: string;
+	titleClass: string;
+	containerClassTotal: string;
+	containerClassInputLabels: string;
+	containerClassValues: string;
+	labelClass: string;
+	isRequired: boolean;
+}
+
+export default function SimpleRadio({
+	name,
+	values,
+	title,
+	titleClass,
+	containerClassTotal,
+	containerClassInputLabels,
+	containerClassValues,
+	labelClass,
+	isRequired,
+}: TextfieldProps) {
+	return (
+		<div className={containerClassTotal}>
+			<p className={titleClass}>
+				{`${title} `}
+				{isRequired && <RequiredAsterisk />}
+			</p>
+			<div className={containerClassValues}>
+				{values.map((value, i) => {
+					return (
+						<div
+							key={`${name}-${i}`}
```

Comment:
Suggestion: if all of the values are unique, we don't need to rely on the index for the key.

Reply:
I was thinking that since technically nothing restricts the values from being repeated (such as having the same id), I personally don't see why using index for key is necessarily a bad idea, but I'd be interested in more elaboration

Reply:
See the React documentation on [rendering lists](https://react.dev/learn/rendering-lists).

> You might be tempted to use an item’s index in the array as its key. In fact, that’s what React will use if you don’t specify a key at all. But the order in which you render items will change over time if an item is inserted, deleted, or if the array gets reordered. Index as a key often leads to subtle and confusing bugs.

Reply:
hm, then it sounds like a more randomly-generated key would be preferrable (like the crypto.randomUUID() the doc mentioed)? But it would be a bit difficult to have them not change on render.

I don't really know what's the best solution here to be honest, I guess in the context that we're using this component we can definitely use the name / id of values though since I'm not sure how the random key would work (and it seems that indices are kind of just a fallback option that we should use when we have nothing else)

Reply:
Random keys also are not a good idea. From the same section of the docs:

> Similarly, do not generate keys on the fly, e.g. with key={Math.random()}. This will cause keys to never match up between renders, leading to all your components and DOM being recreated every time. Not only is this slow, but it will also lose any user input inside the list items. Instead, use a stable ID based on the data.

You've set all the names to be unique and they should be that way, or else we wouldn't be able to process the form properly, so you can just remove the index part (the `-${i}`) from the key.

Reply:
Given the purpose of this (and similar) components is for display options of form elements, it wouldn't make sense for the consumer to provide duplicate values. Using the value as the key would also report such a mistake which could go unnoticed if the index were combined.